### DISCUSSION TITLE
I made a script to run this program across MANY channels almost fully automated

### DISCUSSION THREAD

**Discussion by ziogref** [ThioJoe/YT-Spammer-Purge#827] [General]
Before I start, this is a Linux tool only.

So I made this tool to take in all your youtube subscriptions and run YT-Spammer-Purge spam tool over it. Its designed to run in a spam report mode but in theory could be used to delete comments in other modes. 

So its built using 6x API projects. 5 following Thio's guide then 1 more that it just an API project key that you feed my script so it can split up the channels across the 5 projects allowing you 50,000 api calls instead of 10,000. I can feed all ~102 of the people I am subscribed to, to scan their latest video, once a day using about 30,000 api calls. 

You can also run this tool in a "headless" mode where you create a cron job say once a day. It will scan all the latest video and sort them into groups by guessing how many API calls will be used and then sorting into groups. This way if a youtuber has a contest or something that makes 10x more comments it can move the API groups around making sure you dont blow out your daily api calls.

If you do choose to run it automated all you need to do is update the token.pickle files for each project once a week.

I am a complete noob when it comes to Github and bash and this is my first bash script (that is more than 3 lines) and my first github project. Also never written a mark down file, so the README.md is very basic at this time.

But anyway check it out over at https://github.com/ziogref/Mass-YT-Spammer

Sorry if this breaks any rules, so please remove if I am, I made this script for myself first and then modified it for general use. Also this DOES NOT modify Thio's project AT ALL. all it does is feed the YT-Spammer-Purge folder a config file, client_secrets.json and token.pickle and then runs "python3 YTSpammerPurge.py" all that have taken is the "SpamPurgeConfig.ini" entered some details and saved it as a template, however I can remove it and just provide instructions on what values to enter where.

**Comment by ikkysleepy** [comment]
Having multiple APIs for the same project does go against YouTube's Policy, but good job.  The code is not for production, so I don't think YouTUbe would do anything about it.  I am curious, what's the goal here? Is the goal to automatically remove all spammers by re-scanning all comments every day?

**Reply by ziogref** [reply]
I am aware that its was most likely breaking YT's TOS but you are right, I'm small enough not really to matter. 

As I said the code was built for my use and adapted for the community from something I do personally. 

The goal? Mass report spam comments across everyone I am subscribed to. I wrote a PowerShell script that split the youtubers I am subscribed to and then on my linux box run a bash script (that the powershell script wrote) to run YT Spammer across them.  I had some cases of (for example Explosm, or Primitive Technology) just having a mass explosion of comments on their video blowing out 1 of my 2 project API Quotas. since I was using windows to make the bash shell it got super gross pretty quick. So wanting to learn a bit of bash I smashed this out in 20 days, not knowing anything about bash. So goal 1, write a bash script, goal 2, make a completely automated script to mass report YT spam. As Thio (and LTT) said, reporting spam will hopefully improve the youtube's automatic spam detector to make this application (and my script) completely redundant. If I just have to update my tokens once a week to help with that, easy.

I run my script, just after my API tokens get reset, once I day, but I'm thinking I might have enough allowance to run it twice a day now that I have 5 projects, in just report mode since thats all I can do

**Comment by TechStudent10** [comment]
Whoa this is so cool!
